test/testppx.md

```diff
@@ -1,84 +1,142 @@
+# Integers
+```ocaml
+# 0z ;;
 - : Z.t = 0
+# 1z ;;
 - : Z.t = 1
+# -1z ;;
 - : Z.t = -1
+# 123z ;;
 - : Z.t = 123
+# -0b010z ;;
 - : Z.t = -2
+# 0o147z ;;
 - : Z.t = 103
+# -0O123z ;;
 - : Z.t = -83
+# 0x18z ;;
 - : Z.t = 24
+# -0Xaaz ;;
 - : Z.t = -170
+
+# 4294967297z (* 2^32+1 *);;
 - : Z.t = 4294967297
+# 0x100000001z ;;
 - : Z.t = 4294967297
+# -4294967297z ;;
 - : Z.t = -4294967297
+# 18446744073709551617z (* 2^64+1 *);;
 - : Z.t = 18446744073709551617
+# 0x10000000000000001z ;;
 - : Z.t = 18446744073709551617
+# -18446744073709551617z ;;
 - : Z.t = -18446744073709551617
+# 0x8fffffffz ;;
 - : Z.t = 2415919103
+
+# 0q ;;
 - : Q.t = 0
+# 1q ;;
 - : Q.t = 1
+# -1q ;;
 - : Q.t = -1
+# 123q ;;
 - : Q.t = 123
+# -0b010q ;;
 - : Q.t = -2
+# 0o147q ;;
 - : Q.t = 103
+# -0O123q ;;
 - : Q.t = -83
+# 0x18q ;;
 - : Q.t = 24
+# 0xaaq ;;
 - : Q.t = 170
+
+# 4294967297q (* 2^32+1 *);;
 - : Q.t = 4294967297
+# 0x100000001q ;;
 - : Q.t = 4294967297
+# -4294967297q ;;
 - : Q.t = -4294967297
+# 18446744073709551617q (* 2^64+1 *);;
 - : Q.t = 18446744073709551617
+# 0x10000000000000001q ;;
 - : Q.t = 18446744073709551617
+# -18446744073709551617q ;;
 - : Q.t = -18446744073709551617
+# 0x8fffffffq ;;
 - : Q.t = 2415919103
-Line 3, characters 0-4:
-3 | 0b2z ;;
-    ^^^^
+```
+
+## Fail
+<!-- $MDX version>=4.08 -->
+```ocaml
+# 0b2z ;;
+Line 1, characters 1-5:
 Error: Invalid literal 0b2z
-Line 1, characters 0-4:
-1 | 0O8z ;;
-    ^^^^
+# 0O8z ;;
+Line 1, characters 1-5:
 Error: Invalid literal 0O8z
-Line 1, characters 0-7:
-1 | 0x12g8z ;;
-    ^^^^^^^
+# 0x12g8z ;;
+Line 1, characters 1-8:
 Error: Invalid literal 0x12g8z
-Line 1, characters 0-4:
-1 | 0b2q ;;
-    ^^^^
+# 0b2q ;;
+Line 1, characters 1-5:
 Error: Invalid literal 0b2q
-Line 1, characters 0-4:
-1 | 0O8q ;;
-    ^^^^
+# 0O8q ;;
+Line 1, characters 1-5:
 Error: Invalid literal 0O8q
-Line 1, characters 0-7:
-1 | 0x12g8q ;;
-    ^^^^^^^
+# 0x12g8q ;;
+Line 1, characters 1-8:
 Error: Invalid literal 0x12g8q
-Line 3, characters 18-25:
-3 | let foo = [ Q.neg 0x12.4q ] ;;
-                      ^^^^^^^
+```
+
+### Error positions
+<!-- $MDX version>=4.08 -->
+```ocaml
+# let foo = [ Q.neg 0x12.4q ] ;;
+Line 1, characters 19-26:
```

Comment:
Small sadness here: we don't have the `^^^` thingy to point the position visually. Is it possible to bring it back? It's not terribly essential otherwise.

Reply:
Seems like mdx doesn't have an option for that, possibly because in interactive display, the way `TERM=dumb ocaml` does it, duplicates the input line to quote it, e.g.
```
# let x = 1.2z;;
Line 1, characters 8-12:
1 | let x = 1.2z;;
            ^^^^
Error: Unknown modifier 'z' for literal 1.2z
```
(just an example without zarith-ppx loaded).
This wasn't apparent before because input and (expected) output were in different files.

Reply:
Maybe a nice feature request to make to mdx :)